File: android/domain/src/main/java/com/reservationapp/domain/model/Reservation.kt
package com.reservationapp.domain.model

data class Reservation(
    val id: String,
    val reservationNumber: String,
    val userId: String,
    val userName: String,
    val restaurantId: String,
    val restaurantName: String,
    val menuId: String,
    val menuName: String,
    val mealTimeSlotId: Int,
    val mealTimeSlotName: String,
    val date: String,
    val appetizer: Boolean,
    val status: String,
    val createdAt: String,
    val updatedAt: String?
)

File: ios/ReservationApp/ReservationApp/Domain/Models/Reservation.swift
//
//  Reservation.swift
//  ReservationApp
//
//  Created by Mehmet Ali Kısacık on 13.11.2025.
//

import Foundation

struct Reservation: Codable, Identifiable {
    let id: String // UUID
    let reservationNumber: String
    let userId: String
    let userName: String
    let restaurantId: String
    let restaurantName: String
    let menuId: String
    let menuName: String
    let menuDate: String // ISO date string
    let mealTimeSlotId: Int
    let mealTimeSlotName: String
    let date: String // ISO date string
    let appetizer: Bool
    let status: String // "Pending", "Active", "Cancelled"
    let createdAt: String // ISO date string
    let updatedAt: String? // ISO date string
}



